File: 110. Balanced Binary Tree.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode(int x) { val = x; }
 * }
 */
class Solution {
    public boolean isBalanced(TreeNode root) {
                
        if (root == null) return true;
        
        else{
            
            if (isBalanced(root.left))

            
            if (Math.abs(height(root.left)-height(root.right)) > 1) return false;
            
            else{
                if (isBalanced(root.left)&&isBalanced(root.right)) return true;
            }
            
                return false;
        }
    }
    
    private int height (TreeNode root){
        if (root == null){            
            return 0;
        }
        
        else{
            
            int l_h = height(root.left);
            int r_h = height(root.right);
            
            if (l_h > r_h) return l_h + 1;
            else return r_h + 1;   
        }
    }
}

File: 110-balanced-binary-tree/110-balanced-binary-tree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def isBalanced(self, root: Optional[TreeNode]) -> bool:
        if root is None:
            return True 
        
        left = self.getHeight(root.left)
        right = self.getHeight(root.right)
        
        return (abs(left - right) <= 1) and self.isBalanced(root.left) and self.isBalanced(root.right)
        
    
    def getHeight(self, root):
        if root is None:
            return -1 # No Nodes - 1 = -1 
        
        l = self.getHeight(root.left)
        r = self.getHeight(root.right)
        
        return max(l, r) + 1
        